schematic/store/synapse.py

```diff
@@ -680,105 +691,47 @@ def getFilesInStorageDataset(
+        dataset_clause = f"({dataset_clause})"
 
-        prefix = walk_back_to_project(
-            current_location=current_entity_location,
-            location_prefix="",
-            skip_entry=True,
-        )
+        # When querying, only include files to exclude entity files and subdirectories
+        where_clauses = [dataset_clause, "type='file'"]
 
-        project_id = self.getDatasetProject(datasetId)
-        project = self.synapse_entity_tracker.get(
-            synapse_id=project_id, syn=self.syn, download_file=False
-        )
-        project_name = project.name
-        file_list = []
+        # Requery the fileview to specifically get the files in the given dataset
+        self.query_fileview(columns=["id", "path"], where_clauses=where_clauses)
 
-        # iterate over all results
-        for dirpath, _, path_filenames in walked_path:
-            # iterate over all files in a folder
-            for path_filename in path_filenames:
-                if ("manifest" not in path_filename[0] and not fileNames) or (
-                    fileNames and path_filename[0] in fileNames
-                ):
-                    # don't add manifest to list of files unless it is specified in the
-                    # list of specified fileNames; return all found files
-                    # except the manifest if no fileNames have been specified
-                    # TODO: refactor for clarity/maintainability
-
-                    if fullpath:
-                        # append directory path to filename
-                        if dirpath[0].startswith(f"{project_name}/"):
-                            path_without_project_prefix = (
-                                dirpath[0] + "/"
-                            ).removeprefix(f"{project_name}/")
-                            path_filename = (
-                                prefix + path_without_project_prefix + path_filename[0],
-                                path_filename[1],
-                            )
-                        else:
-                            path_filename = (
-                                prefix + dirpath[0] + "/" + path_filename[0],
-                                path_filename[1],
-                            )
+        # Exclude manifest files
+        non_manifest_files = self.storageFileviewTable.loc[
+            ~self.storageFileviewTable["path"].str.contains("synapse_storage_manifest"),
+            :,
+        ]
+
+        # Remove all files that are not in the list of fileNames
+        if fileNames:
+            filename_regex = "|".join(fileNames)
+
+            matching_files = non_manifest_files["path"].str.contains(
+                filename_regex, case=False, regex=True
+            )
+
+            non_manifest_files = non_manifest_files.loc[matching_files, :]
+
+        # Truncate path if necessary
+        if not fullpath:
+            non_manifest_files.path = non_manifest_files.path.apply(os.path.basename)
```

Comment:
Since I know schematic will always write these with a forward slash I also double checked this logic in linux & windows to verify. It passed!

```
import os
import pandas as pd

# Build a df with a `path` column
df = pd.DataFrame(
    {"path": ["/tmp/tmpnin5mnj1/1", "/tmp/tmpnin5mnj1/2", "/tmp/tmpnin5mnj1/3"]})
print(df.path)
df.path = df.path.apply(os.path.basename)
print(df.path)
```
Result:
```
0    /tmp/tmpnin5mnj1/1
1    /tmp/tmpnin5mnj1/2
2    /tmp/tmpnin5mnj1/3
Name: path, dtype: object
0    1
1    2
2    3
Name: path, dtype: object
```

Reply:
@BryanFauble thanks!